app/src/main/java/com/vero/tasky/agenda/presentation/agendaevent/components/AttendeesComponent.kt

```diff
@@ -0,0 +1,54 @@
+package com.vero.tasky.agenda.presentation.agendaevent.components
+
+import androidx.compose.foundation.layout.*
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import com.vero.tasky.R
+import com.vero.tasky.agenda.domain.model.Attendee
+import com.vero.tasky.agenda.presentation.agendaevent.model.Filter
+
+@Composable
+fun AttendeesComponent(
+    attendees: List<Attendee>,
+    isEditable: Boolean,
+    userId: String,
+    modifier: Modifier = Modifier
+) {
+    var currentFilter by remember { mutableStateOf(Filter.ALL) }
+
+    Spacer(modifier = Modifier.height(32.dp))
+    Row(
+        verticalAlignment = Alignment.CenterVertically,
+        horizontalArrangement = Arrangement.SpaceBetween,
+        modifier = modifier.fillMaxWidth()
+    ) {
+        Filter.values().forEach {
+            VisitorFilterButton(
+                filter = it,
+                isSelected = it == currentFilter,
+                onFilterClick = { filter -> currentFilter = filter })
+        }
+    }
+    Spacer(modifier = Modifier.height(16.dp))
+    if (attendees.any { it.isGoing } &&
```

Comment:
You are filtering on each recomposition
This can cause a performance issue

I recommend directly applying the filter on the viewmodel (so having the "currentFilter" in the vm also)